go/consensus/tendermint/config/config.go

```diff
@@ -0,0 +1,170 @@
+// Package config implements global configuration options.
+package config
+
+import (
+	"fmt"
+	"time"
+)
+
+// Config is the Tendermint configuration structure.
+type Config struct {
+	// Tendermint mode (full, archive).
+	Mode string `yaml:"mode"`
+
+	// Tendermint address advertised to other nodes.
+	ExternalAddress string `yaml:"external_address"`
+	// Tendermint listen address.
+	ListenAddress string `yaml:"listen_address"`
+
+	// Max number of inbound peers.
+	P2PMaxNumInboundPeers int `yaml:"p2p_max_num_inbound_peers"`
+	// Max number of outbound peers (excluding persistent peers).
+	P2PMaxNumOutboundPeers int `yaml:"p2p_max_num_outbound_peers"`
+	// Rate at which P2P packets can be sent (bytes/s).
+	P2PSendRate int64 `yaml:"p2p_send_rate"`
+	// Rate at which P2P packets can be received (bytes/s).
+	P2PRecvRate int64 `yaml:"p2p_recv_rate"`
```

Comment:
Yeah, I have a comment about that in the P2P code...  These are just translated from the original code.  I'll change them to unsigned.

Reply:
Oh, I see why these are signed... because tendermint also uses signed types for these :man_facepalming:
We just pass them directly, so in this case I guess it's better to leave them signed in this config and just check that they're >= 0, like I've done in `Validate()`.